src/main/kotlin/snyk/amplitude/api/AmplitudeExperimentApiClient.kt

```diff
@@ -0,0 +1,81 @@
+package snyk.amplitude.api
+
+import com.intellij.openapi.diagnostic.logger
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import snyk.net.HttpClient
+import java.io.IOException
+
+class AmplitudeExperimentApiClient private constructor(
+    private val baseUrl: String,
+    private val httpClient: OkHttpClient
+) {
+    private val log = logger<AmplitudeExperimentApiClient>()
+    private lateinit var retrofit: Retrofit
+    private lateinit var variantServiceEndpoint: VariantService
+
+    /**
+     * Fetches all Amplitude experiments belongs to the user.
+     */
+    @Suppress("TooGenericExceptionCaught")
+    fun allVariants(user: ExperimentUser): Map<String, Variant> {
+        if (user.userId.isBlank()) {
+            log.warn("user id is empty; amplitude may not resolve identity")
+        }
+        log.debug("Fetch variants for user: $user")
+
+        return try {
+            val response = variantService().sdkVardata(user).execute()
+            if (!response.isSuccessful) {
+                log.info("Error response: $response")
+                return emptyMap()
+            }
+
+            val variants = response.body()
+            log.debug("Received variants: $variants")
+            variants ?: emptyMap()
+        } catch (e: IOException) {
+            log.error("Could not fetch variants because of network communication error with amplitude server", e)
```

Comment:
log.error will be shown to the user as exception

Reply:
good catch!